router.go

```diff
@@ -187,6 +188,27 @@ func (p *ControllerRegister) addWithMethodParams(pattern string, c ControllerInt
 	route.methods = methods
 	route.routerType = routerTypeBeego
 	route.controllerType = t
+	route.initialize = func() ControllerInterface {
```

Comment:
I am still thinking about this? As if add this initialize, the cost would be high? Could you write a bench test for this?